src/service/notification_service/test_utils.rs

```diff
@@ -0,0 +1,72 @@
+use super::{
```

Comment:
Do you think it might make sense to use mockall (https://crates.io/crates/mockall) for these mocks as well for consistency? (I introduce it with the file upload and it works rather nicely so far). It also has an API for introspecting calls etc.

If not, I'm also totally fine with having some auto-mocking and some manual, fine-tuned mocks for specific testing :+1: